public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/reducer.ts

```diff
@@ -80,7 +80,7 @@ export const queriesAndExpressionsReducer = createReducer(initialState, (builder
       const query = payload.recordingRuleQueries[0];
       const recordingRuleQuery = {
         ...query,
-        ...{ expr: payload.expression, model: { expr: payload.expression, refId: query.model.refId } },
+        ...{ expr: payload.expression, model: payload.recordingRuleQueries[0].model },
```

Comment:
Are we sure the `recordingRuleQueries` array is not empty?

Reply:
I think it's safest to use optional chaining here (and this has also been defined as `query` variable above so no need for the entire payload path here)